Lib/test/test_httpservers.py

```diff
@@ -539,6 +539,57 @@ def test_get(self):
             finally:
                 os.chmod(self.tempdir, 0o755)
 
+    def test_range_get(self):
+        route = self.base_url + '/test'
+        response = self.request(route)
+        self.assertEqual(response.getheader('accept-ranges'), 'bytes')
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        # valid ranges
+        response = self.request(route, headers={'Range': 'bYtEs=3-12'})  # case insensitive
+        self.assertEqual(response.getheader('content-range'), 'bytes 3-12/30')
+        self.assertEqual(response.getheader('content-length'), '10')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[3:13])
+
+        response = self.request(route, headers={'Range': 'bytes=3-'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 3-29/30')
+        self.assertEqual(response.getheader('content-length'), '27')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[3:])
+
+        response = self.request(route, headers={'Range': 'bytes=-5'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 25-29/30')
+        self.assertEqual(response.getheader('content-length'), '5')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[25:])
+
+        response = self.request(route, headers={'Range': 'bytes=29-29'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 29-29/30')
+        self.assertEqual(response.getheader('content-length'), '1')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[29:])
+
+        # end > file size
+        response = self.request(route, headers={'Range': 'bytes=25-100'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 25-29/30')
+        self.assertEqual(response.getheader('content-length'), '5')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[25:])
+
+        # invalid ranges
+        response = self.request(route, headers={'Range': 'bytes=100-200'})
+        self.assertEqual(response.getheader('content-range'), 'bytes */30')
+        self.check_status_and_reason(response, HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+        response = self.request(route, headers={'Range': 'bytes=4-3'})
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        response = self.request(route, headers={'Range': 'bytes=wrong format'})
```

Comment:
Can you also test non-ASCII Unicode decimals? see: https://www.compart.com/en/unicode/category/Nd. Also we can add some more cases such as `bytes=` and `bytes=--` (to exercise "almost valid" formats). It's a good way to check that the regex stays stable in the future.